flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/MatchCodeGenerator.scala

```diff
@@ -118,14 +145,19 @@ class MatchCodeGenerator(
     GeneratedFunction(funcName, returnType, funcCode)
   }
 
-  def generatePartitionKeyAccess(
+  /**
+    * Extracts partition keys from any element of the match
+    *
+    * @param partitionKey partition key to be extracted
+    * @return generated code for the given key
+    */
+  private def generatePartitionKeyAccess(
     partitionKey: RexInputRef
   ) : GeneratedExpression = {
-    // extracts partition keys from any element of the match
     val eventNameTerm = newName("event")
     val eventTypeTerm = boxedTypeTermForTypeInfo(input)
 
-    val keyAccess = generateFieldAccess(input, eventNameTerm, partitionKey.getIndex)
+    val keyAccess = generateNullableInputFieldAccess(input, eventNameTerm, partitionKey.getIndex)
```

Comment:
Can the input be nullable? Otherwise we can use a different method that does not check for null.